pkg/granted/registry/add.go

```diff
@@ -0,0 +1,197 @@
+package registry
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"strings"
+
+	grantedConfig "github.com/common-fate/granted/pkg/config"
+	cfflags "github.com/common-fate/granted/pkg/urfav_overrides"
+
+	"github.com/urfave/cli/v2"
+)
+
+// Prevent issues where these flags are initialised in some part of the program then used by another part
+// For our use case, we need fresh copies of these flags in the app and in the assume command
+// we use this to allow flags to be set on either side of the profile arg e.g `assume -c profile-name -r ap-southeast-2`
+func GlobalFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.StringFlag{Name: "ref", Aliases: []string{"r"}, Usage: "Used to reference a specific commit hash, tag name or branch name"},
+	}
+}
+
+var AddCommand = cli.Command{
+	Name:  "add",
+	Flags: GlobalFlags(),
```

Comment:
It is simpler to just put the flags directly here on the command (it makes it a lot easier to see what flags relate to a particular command, rather than needing to look up what `GlobalFlags` is)

```suggestion
	Flags: []cli.Flag{
 		&cli.StringFlag{Name: "ref", Aliases: []string{"r"}, Usage: "Used to reference a specific commit hash, tag name or branch name"},
 	}
```